### PR TITLE
Install Redash on OCDSData server

### PR DESCRIPTION
**Pull request by odscjames**
https://github.com/open-contracting/ocdsdata/issues/92

Also fix createdatabase - the CLI had changed.

### CODE REVIEW

File: salt/redash/upgrade-nointeraction [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+import urllib
+import argparse
+import os
+import subprocess
+import sys
+from collections import namedtuple
+from fnmatch import fnmatch
+
+import requests
+
```
Comment (BibianaC) [inline salt/redash/upgrade-nointeraction]: Would it be possible to order imports? Pycharm does it for you ;-) 'code/optimize imports'

File: salt/redash/upgrade-nointeraction [hunk 2]
```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (odscjames) [reply salt/redash/upgrade-nointeraction]: Yes - will do

[the remark this answers is not in the document]
Comment (odscjames) [reply salt/redash/upgrade-nointeraction]: New commit pushed

File: salt/redash/upgrade-nointeraction (hunk 1 above)
Comment (odscjames) [reply salt/redash/upgrade-nointeraction]: Added comments. It's not quite a full diff, but hopefully it's clear enough (and the source may change anyway, so I think this is better than a full diff)

**Comment by odscjames** [comment]
Added comments, review requested again please :-)

File: salt/redash/upgrade-nointeraction (hunk 1 above)
Comment (BibianaC) [reply salt/redash/upgrade-nointeraction]: @odscjames sorry, I have just seen your message. I don't understand your answer. That you have copy/pasted something doesn't mean that you cannot fix something that is not ok. Would you leave a bug in a copy/paste code? I don't think so.

Comment (BibianaC) [reply salt/redash/upgrade-nointeraction]: Thanks for your explanation @odscjames .

File: salt/ocdsdata.sls [hunk 3]
```diff
@@ -67,9 +69,47 @@ ocdsdata-prerequisites  :
...
+    - require:
+      - file: /etc/nginx/redash-htpasswd
+      - file: /etc/nginx/sites-available/redash
+
+/opt/redash/current/bin/upgrade-nointeraction:
+  file.managed:
+    - source: salt://redash/upgrade-nointeraction
+    - mode: 744
+    - require:
+      - cmd: run-redash-upgrade-nointeraction
```
[the remark this answers is not in the document]
Comment (Bjwebb) [reply salt/ocdsdata.sls]: The require.

[the remark this answers is not in the document]
Comment (odscjames) [reply salt/ocdsdata.sls]: Without it there is a chance that on first install, it places the upgrade-nointeraction file before installing the software, then it tries to run the upgrade script when there is no software to upgrade, and this would probably error. (Salt decieding to run the upgrade files uses the existance of the upgrade file as a trigger)

I admit I'm now not sure if the original reason I put that there was mistaken, but still think it's good for that reason.

**Comment by odscjames** [comment]
Last commit not tested yet as having some problems with my local salt.

Also there is a similiar "ocdsdata-redash" branch in the private repo

**Comment by robredpath** [comment]
@odscjames do feel free to spin up a new box at Bytemark to test things if local VM issues are slowing you down

**Comment by odscjames** [comment]
Got it working on local VM - P.R. please! @robredpath @Bjwebb

**Comment by Bjwebb** [comment]
Remember to merge the private pillar branch also.
